src/components/learner-credit-management/data/utils.js

```diff
@@ -680,36 +681,50 @@ export const getAssignableCourseRuns = ({ courseRuns, subsidyExpirationDatetime,
     enrollStart: courseRun.hasEnrollStart ? dayjs.unix(courseRun.enrollStart).toISOString() : null,
     upgradeDeadline: dayjs.unix(courseRun.upgradeDeadline).toISOString(),
   }));
+
   const assignableCourseRunsFilter = ({
-    enrollBy, enrollStart, start, isActive, hasEnrollBy, isLateEnrollmentEligible,
+    enrollBy, enrollStart, start, hasEnrollBy, hasEnrollStart, isActive, isLateEnrollmentEligible,
   }) => {
+    const isEnrollByDateValid = isEnrollByDateWithinThreshold({
+      hasEnrollBy,
+      enrollBy,
+      isLateRedemptionAllowed,
+    });
+    const isStartDateValid = isStartDateWithinThreshold({
+      hasEnrollStart,
+      enrollStart,
+      start,
+      subsidyExpirationDatetime,
+    });
+
     // Determine eligibility based on the provided enrollBy, start, and enrollStart date
-    const isEligibleForEnrollment = isEnrollByDateWithinThreshold(
-      {
-        enrollBy,
-        isLateRedemptionAllowed,
-      },
-    ) && isStartDateWithinThreshold(
-      {
-        enrollStart,
-        start,
-        subsidyExpirationDatetime,
-      },
-    );
+    const isEligibleForEnrollment = isEnrollByDateValid && isStartDateValid;
+
+    if (!isEligibleForEnrollment) {
+      // Basic checks against this content's critical dates and their relation to
+      // the current date and subsidy expiration date have failed.
+      return false;
+    }
     if (hasEnrollBy && isLateRedemptionAllowed && isDateBeforeToday(enrollBy)) {
-      return isLateEnrollmentEligible && isEligibleForEnrollment;
+      // Special case: late enrollment has been enabled by ECS for this budget, and
+      // isEligibleForEnrollment already succeeded, so we know that late enrollment
+      // would be happy given enrollment deadline of the course.  Now all we need
+      // to do is make sure the run itself is generally eligible for late enrollment
+      // (e.g. it is published and not archived, etc.)
```

Comment:
nit: `isLateEnrollmentEligible` does not take into account whether a course run is published. Cross-check the logic abstracted within `isLateEnrollmentEligible` ([source](https://github.com/openedx/enterprise-catalog/blob/4db5daa4c5f7b8172e99174e3e036683c8886da1/enterprise_catalog/apps/catalog/algolia_utils.py#L1255-L1267)). 

```python
def _get_is_late_enrollment_eligible(course_run):
    """
    Determines if the course run is eligible for late enrollment:
      * Must not be archived
      * Must have a marketing URL
      * Must have seats
    """
    is_archived = course_run.get('availability') == 'Archived'
    has_marketing_url = bool(course_run.get('marketing_url'))
    has_seats = bool(course_run.get('seats'))
    if is_archived or not has_marketing_url or not has_seats:
        return False
    return True
```